### PR TITLE
Allow missing logger name in log viewer

### PR DESCRIPTION
**Pull request by zbynek**
Sometimes logger name is missing (e.g. for log records introduced in https://github.com/jenkinsci/jenkins/pull/6044). Such log records could not be displayed because the logger name was compared to previousentry and that comparison failed with a NPE.

### Proposed changelog entries

* Display log entries with missing logger names in the log viewer.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [n/a] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

🤷 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by basil — APPROVED** [review]

File: core/src/main/java/hudson/Functions.java
Comment (jglick) [inline core/src/main/java/hudson/Functions.java:577]: tip: `Util.fixNull`

Comment (zbynek) [reply core/src/main/java/hudson/Functions.java:577]: @jglick thanks, noted for next time. By the way do you know why [these](https://github.com/jglick/jenkins/blob/d3dc69f8b7270e885f0c055b2210706fc5411c96/core/src/main/java/hudson/util/RingBufferLogHandler.java#L106) `<discarded>` log records show up in the log UI? This PR just makes sure they can be displayed without NPEs, but a proper fix would be to make sure they don't show up at all...

Comment (basil) [reply core/src/main/java/hudson/Functions.java:577]: The `<discarded>` log records were introduced in #6044, which was itself an amendment to #6018. See those two PRs for the reasoning behind this change.

**Pull request merged**
